packages/example-log-extension/README.md

```diff
@@ -245,24 +252,27 @@ import {CoreBindings} from '@loopback/core';
 import {OperationArgs, ParsedRequest} from '@loopback/rest';
 import {getLogMetadata} from '../decorators/log.decorator';
 import {EXAMPLE_LOG_BINDINGS, LOG_LEVEL} from '../keys';
-import {LogFn, TimerFn, HighResTime, LevelMetadata} from '../types';
+import {LogAction, TimerFn, HighResTime, LevelMetadata, LogFn} from '../types';
 import chalk from 'chalk';
 
-export class LogActionProvider implements Provider<LogFn> {
+export class LogActionProvider implements Provider<LogAction> {
   constructor(
     @inject.getter(CoreBindings.CONTROLLER_CLASS)
     private readonly getController: Getter<Constructor<{}>>,
     @inject.getter(CoreBindings.CONTROLLER_METHOD_NAME)
     private readonly getMethod: Getter<string>,
     @inject(EXAMPLE_LOG_BINDINGS.APP_LOG_LEVEL)
     private readonly logLevel: number,
+    // Use a getter to defer to resolution of Logger
+    @inject.getter(EXAMPLE_LOG_BINDINGS.LOGGER) public getLogger: Getter<LogFn>,
```

Comment:
Please use property-based injection for dependencies that have reasonable default. That way users instantiating this Provider class directly (typically unit tests) don't have to repeat specification of default argument values (i.e. logToConsole).

See our own docs: http://loopback.io/doc/en/lb4/Dependency-injection.html#property-injection (emphasis is mine)

> Property injection is usually used for optional dependencies which are not required for the class to function or **for dependencies that have a reasonable default.**

Reply:
@raymondfeng I still think we should be using property-based injection here, it's the best practice we are recommending ourselves in our docs! If you disagree then please explain your reasoning.